docs/sc-memory/api/cpp/guides/simple_guide_for_implementing_agent.md

```diff
@@ -0,0 +1,524 @@
+  // should be subscribed to sc-event of adding outgoing sc-arc from sc-element
+  // `action_initiated`. It is default parameter in these method if you want to
+  // subscribe agent class inherited from `ScActionAgent`.
+
+// This way of subscribing agents makes it easier to write code. 
+// You don't have to think about unsubscribing agents after 
+// the system shutdown - your module will do it all by itself.
+```
+
+If you want to initialize something else in your module besides agents, you can override methods `Initialize(ScMemoryContext * context) override;` and `Shutdown(ScMemoryContext * context) override;`.
+
+**sc_set_module.hpp**
+
+```diff
+class ScSetModule : public ScModule
+{
++ void Initialize(ScMemoryContext * context) override;
++ void Shutdown(ScMemoryContext * context) override;
+};
+```
+
+**sc_set_module.cpp**
+
+```diff
+SC_MODULE_REGISTER(ScSetModule)
+  ->Agent<ScAgentCalculateSetPower>();
+
++ // This method will be called once. 
++ void ScSetModule::Initialize(ScMemoryContext * context)
++ {
++   // Implement initialize of your objects here.
++ }
++ // This method will be called once. 
++ void ScSetModule::Shutdown(ScMemoryContext * context)
++ {
++   // Implement shutdown of your objects here.
++ }
+```
+
+### **8. Write tests for implemented agent. Check agent logic.**
+
+To make sure how your agent works it is best to create tests and cover in them all possible cases that your agent has to handle. For this, create a separate file with test cases and implement them.
+
+```diff
+ set-agents-module/
+ ├── CMakeLists.txt
+ ├── agents/
+ │   ├── sc_agent_calculate_set_power.hpp
+ │   └── sc_agent_calculate_set_power.сpp
+ ├── keynodes/
+ │   └── sc_set_keynodes.hpp
++├── tests/
++│   └── test_sc_agent_calculate_set_power.cpp
+ ├── sc_set_module.hpp
+ └── sc_set_module.cpp
+```
+
+**CMakeLists.txt**
+
+```diff
+file(GLOB SOURCES CONFIGURE_DEPENDS
+    "*.cpp" "*.hpp"
+    "agents/*.cpp" "agents/*.hpp"
+    "keynodes/*.cpp" "keynodes/*.hpp"
+)
+
+add_library(set-agents SHARED ${SOURCES})
+target_link_libraries(set-agents LINK_PRIVATE sc-memory)
+target_include_directories(set-agents PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+
++file(GLOB TEST_SOURCES CONFIGURE_DEPENDS
++    "tests/*.cpp"
++)
+
+# Create executable for tests.
++add_executable(set-agents-tests ${TEST_SOURCES})
++target_link_libraries(set-agents-tests LINK_PRIVATE set-agents)
++target_include_directories(set-agents-tests 
++    PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}
++)
+
++add_test(NAME set-agents-tests COMMAND set-agents-tests)
+```
+
+**test_sc_agent_calculate_set_power.cpp**
+
+```cpp
+#include <sc-memory/sc_test.hpp>
+
+#include "agents/sc_agent_calculate_set_power.hpp"
+#include "keynodes/sc_set_keynodes.hpp"
+
+using AgentTest = ScMemoryTest;
+
+TEST_F(AgentTest, AgentCalculateSetPowerFinishedSuccessfully)
+{
+  // Create action with class that your agent performs.
+  ScAction action 
```

Comment:
Unknown type name 'ScAction'
and 55 more errors in this file when I tried to follow this guide

Reply:
Fixed